### ISSUE TITLE
authentication with jwt does not work anymore

### ISSUE THREAD

**Issue by rhwilr** [adonisjs/adonis-framework#756]
After updating from `4.0.28` to `4.0.29` a correctly authenticated request with Bearer token returns `null` when accessing `auth.user`

**Comment by thetutlage** [comment]
Would you mind re-generate the JWT token and then verify?

**Comment by rhwilr** [comment]
Ia am using fresh tokens. I use the following code to generate a token for a user:
```js
const token = await auth
        .withRefreshToken()
        .attempt(email.trim(), password)
```
and use the returned token on subsequent requests. Until now this method returned the current user based on the token:
```js 
async get ({auth}) {
    return auth.user
}
```

**Comment by thetutlage** [comment]
Looking into it

**Comment by thetutlage** [comment]
Seems to be working fine on my end. Can u share the JWT config from `config/auth.js` file?

**Comment by rhwilr** [comment]
Sure.

```js
{
  authenticator: 'jwt',
  jwt: {
    serializer: 'lucid',
    model: 'App/Models/User',
    scheme: 'jwt',
    uid: 'email',
    password: 'password',
    options: {
      secret: 'self::app.appKey'
    }
  }
}
```

**Comment by rhwilr** [comment]
I'm looking through the changes since `4.0.28` and can't find anything related to api or authentication..
But it is definitely related to `"@adonisjs/framework": "^4.0.29"`  because if I go back a version it works again.

**Comment by thetutlage** [comment]
Everything seems to be fine. So there was security bug in the way config values are resolved. I fixed that, but everything seems to be working fine after that at my end.

Lemme do a full diagnose and get back to you

**Comment by thetutlage** [comment]
@rhwilr Do u want to try it once by installing following repos from Github ( develop branch )

```
npm install https://github.com/adonisjs/adonis-framework.git\#develop
npm install https://github.com/adonisjs/adonis-auth.git\#develop
```

**Comment by rhwilr** [comment]
Your awesome! It works :+1:
I guess I will have to change 
```js
   jwt: {
     secret: 'self::app.appKey'
   }
```
as mentioned in your blog post.  Would you recommend changing it to: 
```js
   jwt: {
     secret: Env.get('APP_KEY')
   }
```

**Comment by thetutlage** [comment]
Yes. I am planning to drop `self` reference since it causes performance overhead.

**Comment by thetutlage** [comment]
Published at `@adonisjs/framework@4.0.30` and `@adonisjs/auth@2.0.11`

**Comment by rhwilr** [comment]
You're the best! Thanks for resolving this issue so quickly. :+1:
